src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -159,13 +166,27 @@ protected String requestNewAuthToken(Credential credential) {
       final I lastPrevItem,
       final int count)
       throws IOException {
-
+    
     if (listWasCalled) {
       throw new EOFException();
     }
 
     val buff = new ArrayList<I>(1);
-    buff.add(itemFactory.getItem(path + prefix, 0, 0));
+
+    if (itemFactory instanceof PathItemFactoryImpl) {
+      Properties properties = new Properties();
+      adminClient = KafkaAdminClient.create(properties);
+      ListTopicsResult result = adminClient.listTopics();
+      try {
+        for (String topicName : result.names().get()) {
+          buff.add(itemFactory.getItem(path + prefix + topicName, 0, 0));
+        }
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

Comment:
don't handle the InterruptedException instance, this may cause the hunging
use `throwUnchecked(e)` instead